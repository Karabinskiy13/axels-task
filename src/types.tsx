export type Tag = {
  id: string;
  text: string;
};

export type Image = {
  largeImageURL: string;
  id: number;
  previewURL: string;
  tags: string;
  hits?: [{ largeImageURL: string; id: number; previewURL: string; tags: string }];
};

export type Favorite = {
  id?: number;
  previewURL?: string;
};
