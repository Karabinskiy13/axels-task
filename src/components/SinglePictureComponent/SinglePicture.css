.picture{
display: flex;
flex-direction: column;
justify-content: center;
}
.picture__image{
    max-height: 150px;
}