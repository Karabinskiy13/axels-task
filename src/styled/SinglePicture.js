import styled from 'styled-components';
import { Card, Col } from 'react-bootstrap';

export const Picture = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
export const Columns = styled(Col)`
  margin-top: 20px;
  cursor: pointer;
`;
export const Modal = styled.button`
  border: none;
  background: none;
`;
