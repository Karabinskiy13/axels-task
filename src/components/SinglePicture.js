import React from 'react';

import Card from 'react-bootstrap/Card';
import 'bootstrap/dist/css/bootstrap.min.css';

import { Columns } from '../styled/SinglePicture';

const SinglePicture = ({ picture, showModal }) => (
  <Columns>
    <Card
      className="picture"
      onClick={() => showModal()}
      border="dark"
      style={{ cursor: 'pointer', maxWidth: '200px' }}>
      <Card.Body
        style={{
          textAlign: 'center',
          minHeight: '180px',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center'
        }}>
        <Card.Img
          style={{ maxWidth: '100%', width: 'auto', height: '100%' }}
          className="picture__image"
          src={picture.previewURL}
        />
      </Card.Body>
    </Card>
  </Columns>
);

export default SinglePicture;
