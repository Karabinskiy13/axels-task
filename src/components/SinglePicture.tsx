import React from 'react';

import Card from 'react-bootstrap/Card';

import { Columns } from '../styled/SinglePicture';
import { Image } from '../types';

interface Props {
  picture: Image;
  showModal: (status: boolean) => void;
  largeImageURL?: string;
  hits?: string;
}

const SinglePicture = ({ picture, showModal }: Props) => (
  <Columns>
    <Card
      className="picture"
      onClick={() => showModal(true)}
      border="dark"
      style={{ cursor: 'pointer', maxWidth: '200px' }}>
      <Card.Body
        style={{
          textAlign: 'center',
          minHeight: '180px',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center'
        }}>
        <Card.Img
          style={{ maxWidth: '100%', width: 'auto', height: '100%' }}
          className="picture__image"
          alt="picture__image"
          src={picture.previewURL}
        />
      </Card.Body>
    </Card>
  </Columns>
);

export default SinglePicture;
