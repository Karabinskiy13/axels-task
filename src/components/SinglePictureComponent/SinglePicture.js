import React from 'react';

import Card from 'react-bootstrap/Card';
import Col from 'react-bootstrap/Col';
import 'bootstrap/dist/css/bootstrap.min.css';

import { Picture } from '../../styled/SinglePicture';

const SinglePicture = ({ picture }) => {
  const { previewURL, tags } = picture;
  return (
    <Col>
      <Picture>
        <Card className="picture" border="dark" style={{ width: '18rem' }}>
          <Card.Body>
            <Card.Img
              style={{ maxHeight: '150px' }}
              className="picture__image"
              src={previewURL}></Card.Img>
            <Card.Header className="picture__title">{tags}</Card.Header>
          </Card.Body>
        </Card>
      </Picture>
    </Col>
  );
};

export default SinglePicture;
