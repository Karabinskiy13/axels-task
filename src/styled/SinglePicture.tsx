import styled from 'styled-components';

export const Columns = styled.div`
  display: flex;
  max-width: 300px;
  align-text: center;
  justify-content: center;
  align-items: center;
  min-height: 180px;
  border: 2px solid gray;
  border-radius: 10px;
  margin-left: auto;
  margin-right: auto;
`;
export const Modal = styled.button`
  border: none;
  background: none;
`;

export const Images = styled.img`
  width: 100%;
  height: 100%;
  max-width: 250px;
  max-height: 160px;
`;
