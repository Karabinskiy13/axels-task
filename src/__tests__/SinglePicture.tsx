import React from 'react';
import { jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SinglePicture from '../components/SinglePicture';

import { Image } from '../types';

const singlePicture: Image = {
  id: 1,
  previewURL: '/image.png',
  tags: 'tags',
  largeImageURL: '/'
};

describe('<SinglePicture>', () => {
  test('Should render component', () => {
    const { asFragment } = render(<SinglePicture picture={singlePicture} showModal={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should call a callback when the card is clicked', async () => {
    const showModalCb = jest.fn(() => true);
    const { asFragment } = render(
      <SinglePicture picture={singlePicture} showModal={showModalCb} />
    );
    const user = userEvent.setup();
    await user.click(screen.getByAltText('picture__image'));
    expect(showModalCb).toBeCalledWith(true);
  });

  //   it('Should display image from props', async () => {
  //     const { asFragment } = render(<SinglePicture picture={singlePicture} showModal={jest.fn()} />);

  //     expect(.find('.picture__image').prop('src')).toEqual(singlePicture.previewURL);
  //   });
});
