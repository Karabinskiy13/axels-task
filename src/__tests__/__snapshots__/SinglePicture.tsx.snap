// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePicture> Should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi dztBFZ col"
  >
    <div
      class="picture card border-dark"
      style="cursor: pointer; max-width: 200px;"
    >
      <div
        class="card-body"
        style="text-align: center; min-height: 180px; display: flex; justify-content: center; align-items: center;"
      >
        <img
          alt="picture__image"
          class="card-img picture__image"
          src="/image.png"
          style="max-width: 100%; width: auto; height: 100%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
